front/src/modules/timesStops/helpers/computeMargins.ts

```diff
@@ -1,41 +1,53 @@
 import type { TrainScheduleResult } from 'common/api/osrdEditoastApi';
-import type { TrainScheduleWithDetails } from 'modules/trainschedule/components/Timetable/types';
+import type {
+  TheoreticalMarginsRecord,
+  TrainScheduleWithDetails,
+} from 'modules/trainschedule/components/Timetable/types';
 import { ms2sec } from 'utils/timeManipulation';
 
 import { formatDigitsAndUnit } from './utils';
+import type { ScheduleEntry } from '../types';
 
-function getTheoreticalMargin(selectedTrainSchedule: TrainScheduleResult, pathStepId: string) {
-  if (selectedTrainSchedule.path.length === 0) {
+export function getTheoreticalMargins(selectedTrainSchedule: TrainScheduleResult) {
+  const selectedTrainScheduleMargins = selectedTrainSchedule.margins;
```

Comment:
You can extract directly margins like this
```suggestion
  const { margins } = selectedTrainSchedule;
```

Reply:
Sure